File: app/src/main/java/com/example/fondodepantalla/Carga.java
package com.example.fondodepantalla;

import androidx.appcompat.app.AppCompatActivity;
import android.app.AlarmManager;
import android.app.PendingIntent;
import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.Typeface;
import android.os.Bundle;
import android.os.Handler;
import android.widget.ImageView;
import android.widget.TextView;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import java.util.Calendar;

public class Carga extends AppCompatActivity {

    TextView app_name, desarrollador;
    ImageView logoCarga;
    private static final String PREFS_NAME = "AsistenciaPrefs";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.carga);

        // Cambiar la fuente
        String ubicacion = "fuentes/choco_cooky.ttf";
        Typeface tf = Typeface.createFromAsset(getAssets(), ubicacion);

        app_name = findViewById(R.id.app_name);
        desarrollador = findViewById(R.id.desarrollador);
        logoCarga = findViewById(R.id.logo_carga);

        // Cambiar el logo según el mes
        logoCarga.setImageResource(getLogoPorMes());

        final int DURACION = 3000;

        new Handler().postDelayed(() -> {
            // Verificar si hay usuario autenticado
            FirebaseAuth mAuth = FirebaseAuth.getInstance();
            FirebaseUser user = mAuth.getCurrentUser();

            // Guardar última apertura si hay usuario
            if (user != null) {
                SharedPreferences prefs = getSharedPreferences(PREFS_NAME, MODE_PRIVATE);
                SharedPreferences.Editor editor = prefs.edit();
                editor.putLong("ultima_apertura", System.currentTimeMillis());
                editor.apply();

                // Programar recordatorio diario
                programarRecordatorio();
            }

            Intent intent;
            if (user != null) {
                intent = new Intent(Carga.this, MainActivityAdministrador.class);
            } else {
                intent = new Intent(Carga.this, InicioSesion.class);
            }

            startActivity(intent);
            finish();

        }, DURACION);

        app_name.setTypeface(tf);
        desarrollador.setTypeface(tf);
    }

    // Método para obtener logo según mes
    private int getLogoPorMes() {
        int mes = Calendar.getInstance().get(Calendar.MONTH) + 1; // Enero=1
        switch (mes) {
            case 2:  return R.drawable.logofeb;
            case 9:  return R.drawable.logosep;
            case 10: return R.drawable.logooct;
            case 11: return R.drawable.logonov;
            case 12: return R.drawable.logodec;
            default: return R.drawable.logodev;
        }
    }

    // ----------------------- RECORDATORIO DIARIO -----------------------
    private void programarRecordatorio() {
        AlarmManager alarmManager = (AlarmManager) getSystemService(ALARM_SERVICE);
        Intent intent = new Intent(this, RecordatorioReceiver.class);
        PendingIntent pendingIntent = PendingIntent.getBroadcast(
                this, 0, intent, PendingIntent.FLAG_IMMUTABLE
        );

        Calendar calendar = Calendar.getInstance();
        calendar.set(Calendar.HOUR_OF_DAY, 9);
        calendar.set(Calendar.MINUTE, 0);
        calendar.set(Calendar.SECOND, 0);

        if (calendar.getTimeInMillis() < System.currentTimeMillis()) {
            calendar.add(Calendar.DAY_OF_YEAR, 1);
        }

        alarmManager.setRepeating(
                AlarmManager.RTC_WAKEUP,
                calendar.getTimeInMillis(),
                AlarmManager.INTERVAL_DAY,
                pendingIntent
        );
    }
}

File: app/src/main/java/com/example/fondodepantalla/Carga.kt
package com.example.fondodepantalla

import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.airbnb.lottie.compose.*
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import kotlinx.coroutines.delay
import java.util.*
import com.google.accompanist.systemuicontroller.rememberSystemUiController

class Carga : ComponentActivity() {

    private val PREFS_NAME = "AsistenciaPrefs"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            CargaScreen()
        }
    }

    @Composable
    fun CargaScreen() {
        val context = LocalContext.current
        val chocoCooky = FontFamily.Cursive

        // Control de status bar y barra de navegación
        val systemUiController = rememberSystemUiController()
        SideEffect {
            systemUiController.setStatusBarColor(
                color = Color.Transparent,
                darkIcons = false // iconos claros
            )
            systemUiController.setNavigationBarColor(
                color = Color.Transparent,
                darkIcons = false // iconos claros
            )
        }

        // Lottie animación
        val composition by rememberLottieComposition(LottieCompositionSpec.Asset("carga_animacion.json"))
        val progress by animateLottieCompositionAsState(composition, iterations = LottieConstants.IterateForever)

        Box(
            modifier = Modifier.fillMaxSize(),
            contentAlignment = Alignment.Center
        ) {
            // Fondo con PNG
            Image(
                painter = painterResource(id = R.drawable.background_log),
                contentDescription = null,
                contentScale = ContentScale.Crop,
                modifier = Modifier.fillMaxSize()
            )

            // Contenido encima del fondo
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center,
                modifier = Modifier.padding(16.dp)
            ) {

                Image(
                    painter = painterResource(id = getLogoPorMes()),
                    contentDescription = "Logo",
                    modifier = Modifier.size(180.dp)
                )

                Spacer(modifier = Modifier.height(20.dp))

                Text(
                    text = context.getString(R.string.app_name),
                    fontFamily = chocoCooky,
                    fontSize = 24.sp,
                    color = Color.White
                )

                Spacer(modifier = Modifier.height(10.dp))

                Text(
                    text = context.getString(R.string.desarrollador),
                    fontFamily = chocoCooky,
                    fontSize = 18.sp,
                    color = Color.White
                )

                Spacer(modifier = Modifier.height(20.dp))

                LottieAnimation(
                    composition = composition,
                    progress = { progress },
                    modifier = Modifier
                        .height(150.dp)
                        .fillMaxWidth()
                )
            }
        }

        // Lógica de espera y navegación
        LaunchedEffect(Unit) {
            delay(3000L)
            val mAuth = FirebaseAuth.getInstance()
            val user: FirebaseUser? = mAuth.currentUser

            if (user != null) {
                val prefs: SharedPreferences =
                    context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
                prefs.edit().putLong("ultima_apertura", System.currentTimeMillis()).apply()

                programarRecordatorio(context)
            }

            val intent = if (user != null) {
                Intent(context, MainActivityAdministrador::class.java)
            } else {
                Intent(context, InicioSesion::class.java)
            }
            context.startActivity(intent)
            (context as ComponentActivity).finish()
        }
    }

    private fun getLogoPorMes(): Int {
        val mes = Calendar.getInstance().get(Calendar.MONTH) + 1
        return when (mes) {
            2 -> R.drawable.logofeb
            9 -> R.drawable.logosep
            10 -> R.drawable.logooct
            11 -> R.drawable.logonov
            12 -> R.drawable.logodec
            else -> R.drawable.logodev
        }
    }

    private fun programarRecordatorio(context: Context) {
        val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
        val intent = Intent(context, RecordatorioReceiver::class.java)
        val pendingIntent = PendingIntent.getBroadcast(
            context,
            0,
            intent,
            PendingIntent.FLAG_IMMUTABLE
        )

        val calendar = Calendar.getInstance()
        calendar.set(Calendar.HOUR_OF_DAY, 9)
        calendar.set(Calendar.MINUTE, 0)
        calendar.set(Calendar.SECOND, 0)

        if (calendar.timeInMillis < System.currentTimeMillis()) {
            calendar.add(Calendar.DAY_OF_YEAR, 1)
        }

        alarmManager.setRepeating(
            AlarmManager.RTC_WAKEUP,
            calendar.timeInMillis,
            AlarmManager.INTERVAL_DAY,
            pendingIntent
        )
    }
}
